### ISSUE TITLE
Let's build a beta

### ISSUE THREAD

**Issue by piranna** [NodeOS/NodeOS#146]
Now that almost all the tasks for the [alpha](https://github.com/NodeOS/NodeOS/issues/37) are complete, I think it's time to start thinking and discussing where to work now:

- [ ] UIs
  - [ ] blessed correctly working by default
    - [ ] NodeOS font
  - [ ] HTML renderer (Chromium Embedded Framework)
- [ ] OS flavours
  - [ ] mainframe
  - [ ] workstation
- [ ] alternative kernels
  - [ ] FreeBSD
  - [ ] runtime.js
- [ ] alternative platforms
  - [ ] ARM (raspberry pi)
  - [ ] MIPS
- [ ] splitted modules for each layer
- [ ] tests for all packages

Any other idea? :-)

**Comment by formula1** [comment]
That looks like the important stuff, If I wasn't so swamped I wouldn't mind getting involved with fbdev.

Btw, I found you again. http://stackoverflow.com/users/586382/piranna

**Comment by piranna** [comment]
fbdev is "almost" working, we have it enabled but no way to use it :-P [I have been working to be able to run node-canvas on NodeOS](https://github.com/Automattic/node-canvas/issues/551) (there's a fork with fbdev support) and after fixing some issues with the libraries and dependencies to make the build process standalone, there are some linking problems because GCC guys think [it's not a valid use case to have a shared library that it's itself statically compiled](https://bugzilla.redhat.com/show_bug.cgi?id=214465#c1) ¬¬ There's a workaround hack regarding [rename the library files](https://bugs.launchpad.net/ubuntu/+source/gcc-4.4/+bug/640734/comments/6) that I need to test, but don't know if it would lead to bigger problems in the future... :-/


> Btw, I found you again. http://stackoverflow.com/questions/12746053/pass-large-amounts-of-data-between-web-worker-and-main-thread

Yeah, that one was regarded to my previous project [ShareIt!](https://github.com/ShareIt-project/ShareIt) :-P

**Comment by netlovers** [comment]
Hmm, is there a reason why you want to statically build everything?

**Comment by piranna** [comment]
The recomended way to generate node.js compiled modules is as statically linked shared libraries, so this way they don't depend on external libraries and can be easily installable. Regarding node-canvas this is somewhat more important, since it link against Cairo graphics, that's a huge and complicated one...

**Comment by lite20** [comment]
> network reverse proxy (only http/ws?)

We can check that one off! :smile:

**Comment by piranna** [comment]
Not fully, it keeps to find a way so users can register their own web
servers and allow virtual hosts, and write a module that wraps 'http'
module to doing so transparent...

**Comment by lite20** [comment]
> Not fully, it keeps to find a way so users can register their own web
servers and allow virtual hosts, and write a module that wraps 'http'
module to doing so transparent...

Darn. We're getting there!

**Comment by DeanHristov** [comment]
Hi, everyone. When I run the command "install" in the stdout on my shell throw error?  See the image below
![snapshot15](https://cloud.githubusercontent.com/assets/3283991/11429435/56190116-9480-11e5-9589-55f2af0c2174.png)

Any ideas?

**Comment by lite20** [comment]
@piranna

**Comment by piranna** [comment]
I honestly don't know what's the error here, could you describe it better? Are you talking about that stat lines? Seems the layers are not being correctly build... Are you using npm3? NodeOS build only works with npm2 by the moment (and the change to npm3 will be difficult... :-( ).

**Comment by lite20** [comment]
> Are you using npm3? NodeOS build only works with npm2 by the moment (and the change to npm3 will be difficult... :-( ).

Oh I will attempt again with npm2 at a later date, and throw the results back here.

**Comment by piranna** [comment]
> Oh I will attempt again with npm2 at a later date, and throw the results back here.

:+1:

**Comment by piranna** [comment]
> Could we be able to read/write to ext4?

Yes, that's possible since almost beginning, why do you ask? Do you have something interesting in your mind?

**Comment by m-onz** [comment]
is there any guidance for contributing to the UI aspects? It might be easier to start with an existing window manager: https://www.os-js.org/

Would it be possible to create the UI with something like electron js?

An interesting idea... using os.js or electron expose the desktop UI as a port so you can directly control a running instance easily without needing VNC.

[edit]  I can build this as an NPM package... and that is the better approach. Sorry to bother!

**Comment by SASHIRl** [comment]
opa
